### ISSUE TITLE
GPL Violation: Android 12 kernel sources for various devices have not been released

### ISSUE THREAD

**Issue by ralph950412** [MiCode/Xiaomi_Kernel_OpenSource#2506]
Can we please see the source code of venus-s-oss? The first SKBMIXM had been released back in October 2021, and we’re still waiting for the source code. This is also referring to other devices which already got Android 12 updates. Thanks and kind regards,

https://www.gnu.org/licenses/gpl-violation.html

**Comment by ralph950412** [comment]
Inspired by https://github.com/OnePlusOSS/android_kernel_oneplus_sm8250/issues/24 and https://twitter.com/pankaj_bore/status/1538170247248678914?t=uUdRl_jKZ1Z8pGdZI-Injw&s=19

**Comment by soumya19x** [comment]
> Can we please see the source code of venus-s-oss? The first SKBMIXM had been released back in October 2021, and we’re still waiting for the source code. This is also referring to other devices which already got Android 12 updates. Thanks and kind regards,
> 
> https://www.gnu.org/licenses/gpl-violation.html

Following.

**Comment by manosdoc** [comment]
> 

I've reported on GPL. This is beyond disgrace. They are making money and again and again not releasing sources. Clever business. Copy others work, profit, give nothing back. It's been months since they have updated most devices. Months

**Comment by aerzjee** [comment]
> Can we please see the source code of venus-s-oss? The first SKBMIXM had been released back in October 2021, and we’re still waiting for the source code. This is also referring to other devices which already got Android 12 updates. Thanks and kind regards,
> 
> https://www.gnu.org/licenses/gpl-violation.html

Following.

**Comment by furopi** [comment]
> Can we please see the source code of venus-s-oss? The first SKBMIXM had been released back in October 2021, and we’re still waiting for the source code. This is also referring to other devices which already got Android 12 updates. Thanks and kind regards,
> 
> https://www.gnu.org/licenses/gpl-violation.html

Me 2 follows...

**Comment by vlvaibhav** [comment]
+1

**Comment by willzyx-hub** [comment]
That will be useless Xiaomi doesn't care about us ...

**Comment by kotah81** [comment]
> That will be useless Xiaomi doesn't care about us ...

Not really, they could be reported and get into trouble because of this.

**Comment by zainarbani** [comment]
I mean, look at [this](https://github.com/realme-kernel-opensource?tab=repositories) [this](https://github.com/OnePlusOSS) [this](https://github.com/MotorolaMobilityLLC) [this](https://developer.sony.com/develop/open-devices/downloads/open-source-archives/) & [this](https://opensource.samsung.com/uploadList?menuItem=mobile&classification1=mobile_phone)

**Comment by JIBIN-P** [comment]
Release the sources😮‍💨

**Comment by JamesLiu0802** [comment]
I suspect xiaomi has a dirty secret, after all, it only releases the source code once when it releases a phone. I've been wondering why my next phone is a fucking xiaomi. Disgusting company, Xiaomi.

**Comment by inglar** [comment]
Xiaomi, release the sources!

**Comment by TheMatheusDev** [comment]
+1
Almost 5 moths after Redmi Note 11 (spes/spesn) launch and nothing about kernel. 
I know that they are busy upgrading various devices to android 12, i hope they release the android 12 kernel version to make this wait worth it

**Comment by HelpFreedom** [comment]
Xiaomi is EVIL!

**Comment by kaffuchino233** [comment]
+1

**Comment by willzyx-hub** [comment]
7 months is quite significant though.

**Comment by gribanoveu** [comment]
we need source pls

**Comment by zAlyson** [comment]
Something is off about they not launching the source code.

**Comment by ralph950412** [comment]
> Something is off about they not launching the source code.

I have a different thought now... Maybe the reason behind them not releasing newer source code is actually because they don't have anything new. If I remember correctly, TKBCNXM has the exact same kernel version SKBCNXM & RKBCNXM do.

**Comment by exbreaker990** [comment]
pls at least give us news. almost to QPR2 google and samsung updated their driver to new gpu ar least update the new driver to Poco X3 GT
